pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
+	if d.HasChange("external_oauth_jws_keys_url") {
+		v := expandStringList(d.Get("external_oauth_jws_keys_url").(*schema.Set).List())
+		if len(v) > 0 {
+			urls := make([]sdk.JwsKeysUrl, len(v))
+			for i := range v {
+				urls[i] = sdk.JwsKeysUrl{JwsKeyUrl: v[i]}
+			}
+			set.WithExternalOauthJwsKeysUrl(urls)
 		}
+	}
 
-		err = snowflake.Exec(db, stmt)
-		if err != nil {
-			return fmt.Errorf("error executing alter statement: %w", err)
+	if d.HasChange("external_oauth_rsa_public_key") {
+		set.WithExternalOauthRsaPublicKey(d.Get("external_oauth_rsa_public_key").(string))
+		if v, ok := d.GetOk("external_oauth_rsa_public_key"); ok {
+			set.WithExternalOauthRsaPublicKey2(v.(string))
 		}
+		// else: force new
 	}
 
-	if runUnset {
-		stmt, err := manager.Unset(unsetInput)
-		if err != nil {
-			return fmt.Errorf("couldn't generate unset statement for external oauth integration: %w", err)
+	if d.HasChange("external_oauth_rsa_public_key_2") {
+		if v, ok := d.GetOk("external_oauth_rsa_public_key_2"); ok {
+			set.WithExternalOauthRsaPublicKey2(v.(string))
 		}
+		// else: force new
+	}
 
-		err = snowflake.Exec(db, stmt)
-		if err != nil {
-			return fmt.Errorf("error executing unset statement: %w", err)
+	if d.HasChange("external_oauth_scope_delimiter") {
+		if v, ok := d.GetOk("external_oauth_scope_delimiter"); ok {
+			set.WithExternalOauthScopeDelimiter(v.(string))
+		} else {
+			// TODO(SNOW-1515781): use UNSET
+			set.WithExternalOauthScopeDelimiter(",")
 		}
 	}
 
-	return ReadExternalOauthIntegration(d, meta)
-}
+	if d.HasChange("external_oauth_scope_mapping_attribute") {
+		// this field is required
+		set.WithExternalOauthScopeMappingAttribute(d.Get("external_oauth_scope_mapping_attribute").(string))
+	}
 
-// DeleteExternalOauthIntegration implements schema.DeleteFunc.
-func DeleteExternalOauthIntegration(d *schema.ResourceData, meta interface{}) error {
-	manager, err := snowflake.NewExternalOauthIntegration3Manager()
-	if err != nil {
-		return fmt.Errorf("couldn't create external oauth integration builder: %w", err)
+	if d.HasChange("external_oauth_snowflake_user_mapping_attribute") {
+		// this field is required
+		if v, ok := d.GetOk("external_oauth_snowflake_user_mapping_attribute"); ok {
+			value, err := sdk.ToExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeOption(v.(string))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithExternalOauthSnowflakeUserMappingAttribute(value)
+		}
 	}
 
-	input := &snowflake.ExternalOauthIntegration3DeleteInput{
-		TopLevelIdentifier: snowflake.TopLevelIdentifier{
-			Name: d.Get("name").(string),
-		},
+	if d.HasChange("external_oauth_token_user_mapping_claim") {
+		v := expandStringList(d.Get("external_oauth_token_user_mapping_claim").(*schema.Set).List())
+		claims := make([]sdk.TokenUserMappingClaim, len(v))
+		for i := range v {
+			claims[i] = sdk.TokenUserMappingClaim{
+				Claim: v[i],
+			}
+		}
+		set.WithExternalOauthTokenUserMappingClaim(claims)
```

Comment:
I assume empty set is valid

Reply:
It needs to be force new, good point. Fixed.